src/figuresClasses.ts

```diff
@@ -2,10 +1,113 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function roundDownHundreds(value: number): number {
+  if (value === Math.round(value)) {
+    return value;
+  }
+
+  return Math.floor(value * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  public readonly shape = 'triangle' as Shape;
+
+  constructor(
+    public readonly color: Color,
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    const sides = [a, b, c];
+    const longestSide = Math.max(a, b, c);
+    const sumOfOtherSides = sides.reduce((acc, side) => {
+      if (side === longestSide) {
+        return acc;
+      }
 
-export class Triangle implements Figure {}
+      return acc + side;
+    }, 0);
+    const isCorrectTriangle = longestSide < sumOfOtherSides;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw Error('Should have positive sides');
+    }
+
+    if (!isCorrectTriangle) {
+      throw Error('Failed to create a triangle with current sides');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
```

Comment:
According to the checklist, the 'getArea' function should not accept any parameters. It should only use the object's fields. In this case, the function is correctly using the object's fields, so no change is needed.